src/Build/Utilities/EngineFileUtilities.cs

```diff
@@ -191,8 +191,10 @@ internal static bool FilespecHasWildcards(string filespecEscaped)
             FileMatcher.SearchAction action = FileMatcher.SearchAction.None;
             string excludeFileSpec = string.Empty;
 
-            if (!FilespecHasWildcards(filespecEscaped) ||
-                FilespecMatchesLazyWildcard(filespecEscaped, forceEvaluateWildCards))
+            var noWildcards = !FilespecHasWildcards(filespecEscaped) || FilespecMatchesLazyWildcard(filespecEscaped, forceEvaluateWildCards);
+
+            // It is possible to return original string if no wildcard matches and no entries in Exclude set. 
+            if (noWildcards && excludeSpecsEscaped?.Any() == false)
```

Comment:
This should be:

```suggestion
            if (noWildcards && excludeSpecsEscaped?.Any() != true)
```

to cover both null and and empty `excludeSpecsEscaped`.